Source/WebCore/page/SecurityOrigin.cpp

```diff
@@ -169,18 +169,6 @@ Ref<SecurityOrigin> SecurityOrigin::create(const URL& url)
     return adoptRef(*new SecurityOrigin(url));
 }
 
-inline bool isSafelistedBlobProtocol(const URL& url)
-{
-    if (!url.isValid())
-        return false;
-
-    // FIXME: we ought to assert we're in WebKitLegacy or a web content process as per 263652@main,
-    // except that assert gets hit on certain tests.
-    return url.protocolIsInHTTPFamily()
-        || url.protocolIsFile()
-        || LegacySchemeRegistry::schemeIsHandledBySchemeHandler(url.protocol());
-}
```

Comment:
I don't think we want to remove this and regress on WPT. Additionally, not offering `blob:` URLs with the same authority is also a security win.

The original proposal of conditionally allowing the PDFJS scheme here still seems reasonable.

Reply:
Agreed. This is breaking WPT tests so we can't do it.

> The original proposal of conditionally allowing the PDFJS scheme here still seems reasonable.

I will allow resource:// too. I'm a little surprised we haven't noticed anything breaking due to that yet, but I guess we only use resource:// for subresources nowadays. (We used to use it for PDF.js itself before it moved from Epiphany into WebKit.)

Reply:
Is there a reason we'd want to create `blob:` URLs inside `resource://` documents? `blob:` URLs have been used as an attack vector in other browsers, it's better to keep them locked down.

Reply:
Same reason you want blob URLs to work when using `webkit-pdfjs-viewer://`. Think of `resource://` and `webkit-pdfjs-viewer://` and all other custom protocols as a replacement for HTTP. You still want the web content to function normally regardless of the protocol used to load it.

`resource://` means the content is compiled into the ELF binary rather than sent over the internet, but that's no reason to prevent blob URLs from working. Concrete example of why blob should work: before `webkit-pdfjs-viewer://` existed, Epiphany used `resource://` to load PDF.js and could hypothetically eventually want to do that again. (WebKit on Linux actually treats `webkit-pdfjs-viewer://` basically the same as `resource://`: the PDF.js content is compiled into the ELF binary using GResource and it just gets a special URL scheme.)

Reply:
The problem is that it allows attackers to mint these URLs as well and they end up having the same authority. That's why you generally want to restrict it. So unless we have an actual use case I'd rather not.

Reply:
The use case is you can package web content into the application binary instead of having to install in in separate files on disk. So instead of WebKit using read() syscalls to read, say, 50 different files when loading an application-defined web page with 49 subresources, all of that content is already in the process address space and gets accessed using pointers instead of syscalls. GResource is an implementation detail and shouldn't affect the functionality of blob URLs.